beacon_node/beacon_chain/src/test_utils.rs

```diff
@@ -595,7 +595,10 @@ where
             .execution_block_generator()
     }
 
-    pub fn set_mock_builder(&mut self, beacon_url: SensitiveUrl) -> MockBuilderServer {
+    pub fn set_mock_builder(
+        &mut self,
+        beacon_url: SensitiveUrl,
+    ) -> impl std::future::Future<Output = ()> {
```

Comment:
nit:
```suggestion
    pub async fn set_mock_builder(&mut self, beacon_url: SensitiveUrl)  {
```
and bellow call ` mock_builder_server.await`

Reply:
true! updated

Reply:
`impl Future` actually implicitly has a `'static` lifetime, and switching to `async` changes this, so had to change it back